connector.go

```
@@ -73,18 +75,20 @@ func (c *Connector) connectWithRetries(ctx context.Context, conn dbConnection, k
 			settings: conn.settings,
 		}
 		err = c.connect(conn)
-		if err == nil {
-			return err
-		}
-
-		if !shouldRetry(c.driverSettings.RetryOn, err.Error()) {
-			break
+		if err != nil {
```

This change does not look correct to me - before, if `c.connect(conn)` above was successful the function would correctly return `nil`, indicating success. Now if `c.connect(conn)` is successful it will retry (`c.driverSettings.Retries` times) because it doesn't exit the loop in any way.

I think only two things _need_ to change from the original code:
* Add and use the err and db vars as you do
* Change the function to `return err` at the end instead of `return nil`

There's a couple of things that might be nice to change as well:
* You could change the old line 77 to `break` to make the logic a little more explicit.
* The existing code sleeps one time more than it should if it runs out of retries
* The log message is misleading - it will say "retrying 0 times", "retrying 1 times", etc if there are multiple retries.

---

thanks! i've made the changes, could you take a look again?